src/kv_client/client.rs

```diff
@@ -0,0 +1,335 @@
+                .clone()
+                .kv_batch_get_async_opt(&req, self.call_options()),
+        )
+    }
+
+    pub fn kv_batch_rollback(
+        &self,
+        keys: impl Iterator<Item = Key>,
+        start_version: u64,
+    ) -> impl Future<Output = Result<kvrpcpb::BatchRollbackResponse>> {
+        let mut req = txn_request!(self.region, kvrpcpb::BatchRollbackRequest);
+        req.set_keys(keys.map(|x| x.into()).collect());
+        req.set_start_version(start_version);
+
+        map_errors(
+            "kv_batch_rollback",
+            self.client
+                .clone()
+                .kv_batch_rollback_async_opt(&req, self.call_options()),
+        )
+    }
+
+    pub fn kv_scan_lock(
+        &self,
+        start_key: Key,
+        max_version: u64,
+        limit: u32,
+    ) -> impl Future<Output = Result<kvrpcpb::ScanLockResponse>> {
+        let mut req = txn_request!(self.region, kvrpcpb::ScanLockRequest);
+        req.set_start_key(start_key.into());
+        req.set_max_version(max_version);
+        req.set_limit(limit);
+
+        map_errors(
+            "kv_scan_lock",
+            self.client
+                .clone()
+                .kv_scan_lock_async_opt(&req, self.call_options()),
+        )
+    }
+
+    pub fn kv_resolve_lock(
+        &self,
+        txn_infos: impl Iterator<Item = TxnInfo>,
+        start_version: u64,
+        commit_version: u64,
+    ) -> impl Future<Output = Result<kvrpcpb::ResolveLockResponse>> {
+        let mut req = txn_request!(self.region, kvrpcpb::ResolveLockRequest);
+        req.set_start_version(start_version);
+        req.set_commit_version(commit_version);
+        req.set_txn_infos(txn_infos.map(Into::into).collect());
+
+        map_errors(
+            "kv_resolve_lock",
+            self.client
+                .clone()
+                .kv_resolve_lock_async_opt(&req, self.call_options()),
+        )
+    }
+
+    pub fn kv_gc(&self, safe_point: u64) -> impl Future<Output = Result<kvrpcpb::GcResponse>> {
+        let mut req = txn_request!(self.region, kvrpcpb::GcRequest);
+        req.set_safe_point(safe_point);
+
+        map_errors(
+            "kv_gc",
+            self.client
+                .clone()
+                .kv_gc_async_opt(&req, self.call_options()),
+        )
+    }
+
+    pub fn kv_delete_range(
+        &self,
+        start_key: Key,
+        end_key: Key,
+    ) -> impl Future<Output = Result<kvrpcpb::DeleteRangeResponse>> {
+        let mut req = txn_request!(self.region, kvrpcpb::DeleteRangeRequest);
+        req.set_start_key(start_key.into());
+        req.set_end_key(end_key.into());
+
+        map_errors(
+            "kv_delete_range",
+            self.client
+                .clone()
+                .kv_delete_range_async_opt(&req, self.call_options()),
+        )
+    }
+
+    fn call_options(&self) -> CallOption {
+        CallOption::default().timeout(self.timeout)
+    }
+}
+
+fn map_errors<Resp, RpcFuture>(
```

Comment:
To be honest, I think the way we call this function is not intuitively clear. And the request name is used for stats instead of errors, which is misleading.
```
map_errors("request_name", grpc_result_of_a_future)
```

I think it would better if we can write in some way like this:
```
grpc_result_of_a_future.map_errors().with_stats(request_name)
```

But then perhaps we need a util trait which has no exact meaning. What do you think? @nrc

Reply:
I can rename it, but using a trait is difficult because the transactonal API isn't using `BoxFuture` and `impl Trait` is not supported in traits. Once we are done moving the transactional API to async/await (and the command-oriented design) then we can use a trait.